### PR TITLE
Julia 0.7 fixes

### PR DESCRIPTION
**Pull request by shashi**
Started using `Missings` here instead of DataValues.

Yet to get rid of Nullables.

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,7 +1,8 @@
-julia 0.6
+julia 0.7-alpha
 Compat 0.17
-DataValues
 PooledArrays 0.2.0
+Nullables
+Dates
```
Comment (quinnj) [inline REQUIRE]: You don't want `Dates` here; it belongs in the `Project.toml` for 0.7, but shouldn't be in the 0.6 REQUIRE

File: src/field.jl [hunk 2]
```diff
@@ -2,7 +2,7 @@ import Base.show
 
 export CustomParser, Quoted
 
-using Compat
+using Compat, Nullables
```
Comment (quinnj) [inline src/field.jl]: We should get rid of `Nullables`; we should be able to use `Union{Nothing, T}` in place of `Nullable{T}`.

File: src/guesstype.jl [hunk 3]
```diff
@@ -26,14 +26,14 @@ function guessdateformat(str)
 
     for (typ, df) in dts
         x, len = try
-            tryparsenext_internal(typ, str, 1, endof(str), df)
+            tryparsenext_internal(typ, str, 1, lastindex(str), df)
         catch err
             continue
         end
-        if !isnull(x)
+        if x != nothing
```
Comment (quinnj) [inline src/guesstype.jl]: `!==` is typically better style when comparing w/ potential `nothing` values. I think it's more compiler friendly as well.

File: test/runtests.jl [hunk 4]
```diff
@@ -316,12 +336,11 @@ end
     str = "1970-02-02 02:20:20"
     @test tryparsenext(tok, str, 1, length(str), opts) |> unwrap == (DateTime("1970-02-02T02:20:20"), length(str)+1)
     @test tryparsenext(tok, str*"x", 1, length(str)+1, opts) |> unwrap == (DateTime("1970-02-02T02:20:20"), length(str)+1)
-    @test tryparsenext(tok, str[1:end-3]*"x", 1, length(str)-2, opts) |> failedat == length(str)-2
-    @test tryparsenext(tok, str[1:end-3]*"y", 1, length(str)-2, opts) |> unwrap == (DateTime("1970-02-02T02:20"), length(str)-2)
+   #@test tryparsenext(tok, str[1:end-3]*"x", 1, length(str)-2, opts) |> failedat == length(str)-2
```
Comment (quinnj) [inline test/runtests.jl]: TODO?

File: test/runtests.jl [hunk 5]
```diff
@@ -335,16 +354,16 @@ import TextParse: _csvread
     """
     data = ((["x", "","x","x y","x"],
               ones(5),
-              DataValueArray(ones(5), Bool[0,1,0,0,1]),
-              DataValueArray(ones(Int,5), Bool[0,0,0,1,0])),
+              [1,missing,1,1,missing],
```
Comment (quinnj) [inline test/runtests.jl]: `[1.0,missing,1.0,1.0,missing]`? Just to technically match what it was previously.

File: test/runtests.jl (hunk 4 above)
Comment (shashi) [reply test/runtests.jl]: Yeah, need to hack some Base date parser code for this. (basically it needs to be able to parse even though there are extra characters after the date)

**Comment by davidanthoff** [comment]
This breaks [CSVFiles.jl](https://github.com/davidanthoff/CSVFiles.jl), which is an important part of the [Queryverse.jl](https://github.com/davidanthoff/Queryverse.jl). Any chance we can keep the ability (as an option) to return ``DataValueVector``s? There is probably even some way to do that without TextParse.jl taking a dep on DataValue, if we allow passing some functions that return a new nullable column type (with the default implementation returning a ``Missing`` based implementation).

**Comment by davidanthoff** [comment]
Bump, I would even prefer a tagged version that works on 0.7 that breaks the ``DataValueVector`` story at this point. I could probably work around it in the short term, and it would allow me to finish porting the Queryverse.jl to 0.7. Is there still something that needs to be done here, or is this essentially ready?

I would probably still like to add an option back in to get ``DataValueVector``s, but that could be a future release, and I do think there is a way to do that that doesn't require a dependency on DataValues.jl here.

**Comment by davidanthoff** [comment]
Alright, please first merge https://github.com/JuliaComputing/TextParse.jl/pull/70 into this branch here, and then this PR here.

**Comment by quinnj** [comment]
@shashi, ready to merge here?

**Comment by shashi** [comment]
We can fix the date parsing separately

**Comment by davidanthoff** [comment]
Hurray! @shashi, could you also tag a release? I’m trying to get the full Queryverse stack running on 0.7 by juliacon, but would need a release for that.

**Pull request merged**